spec/models/service_spec.rb

```diff
@@ -732,6 +732,42 @@
     end
   end
 
+  describe '#add_to_service' do
+    let(:service) { FactoryGirl.create(:service) }
+    let(:child_service) { FactoryGirl.create(:service) }
+
+    it 'associates a child_service to the service' do
+      child_service.add_to_service(service)
+
+      expect(service.reload.services).to include(child_service)
+    end
+
+    it 'raise an error if the child_service is already part of a service' do
+      child_service.add_to_service(service)
+
+      expect { child_service.add_to_service(service) }.to raise_error MiqException::Error
+    end
+  end
+
+  describe '#remove_from_service' do
+    let(:service) { FactoryGirl.create(:service) }
+    let(:child_service) { FactoryGirl.create(:service) }
+
+    it 'associates a child_service to the service' do
+      child_service.add_to_service(service)
+      expect(child_service.service).to eq(service)
```

Comment:
Should it be: ```expect(service.services).to eq(child_service)``` ?

Reply:
I'm not sure this one matters, we just want to ensure that the two services were linked so we properly validate the removal logic.  We either check that the child is associated to the single parent object OR check that the parent includes the child service, unless I am missing something here.